### ISSUE TITLE
failed to delete ns 

### ISSUE THREAD

**Issue by stoneshi-yunify** [kubesphere/storageclass-accessor#5]
kubectl delete ns failed, the ns stuck in Terminating.

Maybe should ignore the NotFound error in the accessor.

```
root@i-8xxc9dh3:~# kubectl get ns
NAME                              STATUS        AGE
default                           Active        3h34m
kube-node-lease                   Active        3h34m
kube-public                       Active        3h34m
kube-system                       Active        3h34m
kubesphere-controls-system        Active        3h31m
kubesphere-monitoring-federated   Active        3h31m
kubesphere-monitoring-system      Active        3h31m
kubesphere-system                 Active        3h33m
local-path-storage                Active        3h34m
nn                                Active        9m44s
stone                             Terminating   127m
root@i-8xxc9dh3:~# kubectl get ns stone -o yaml
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    kubesphere.io/creator: admin
  creationTimestamp: "2022-09-27T05:27:22Z"
  deletionTimestamp: "2022-09-27T07:23:46Z"
  labels:
    kubernetes.io/metadata.name: stone
    kubesphere.io/namespace: stone
    kubesphere.io/workspace: system-workspace
  name: stone
  ownerReferences:
  - apiVersion: tenant.kubesphere.io/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: Workspace
    name: system-workspace
    uid: c445aae4-2411-432e-bea2-9871b5e5f1c9
  resourceVersion: "27812"
  uid: 1f2def08-4227-41e9-b7c1-2fdde7814729
spec:
  finalizers:
  - kubernetes
status:
  conditions:
  - lastTransitionTime: "2022-09-27T07:23:54Z"
    message: All resources successfully discovered
    reason: ResourcesDiscovered
    status: "False"
    type: NamespaceDeletionDiscoveryFailure
  - lastTransitionTime: "2022-09-27T07:23:54Z"
    message: All legacy kube types successfully parsed
    reason: ParsedGroupVersions
    status: "False"
    type: NamespaceDeletionGroupVersionParsingFailure
  - lastTransitionTime: "2022-09-27T07:23:54Z"
    message: 'Failed to delete all resource types, 1 remaining: admission webhook
      "storageclass-accessor.storage.kubesphere.io" denied the request: resource name
      may not be empty'
    reason: ContentDeletionFailed
    status: "True"
    type: NamespaceDeletionContentFailure
  - lastTransitionTime: "2022-09-27T07:23:56Z"
    message: All content successfully removed
    reason: ContentRemoved
    status: "False"
    type: NamespaceContentRemaining
  - lastTransitionTime: "2022-09-27T07:23:54Z"
    message: All content-preserving finalizers finished
    reason: ContentHasNoFinalizers
    status: "False"
    type: NamespaceFinalizersRemaining
  phase: Terminating
```

ks-controller-manager log:
```
E0927 15:27:38.172758       1 pvc.go:65] get target Delete PVC from client failed, err:resource name may not be empty
I0927 15:27:44.244422       1 reconciler.go:682] controllers/Helm "msg"="Release reconciled" "nginx"={"Namespace":"kubesphere-controls-system","Name":"kubesphere-router-kubesphere-system-ingress"} "name"="kubesphere-router-kubesphere-system-ingress" "version"=1
I0927 15:28:28.153129       1 reconciler.go:682] controllers/Helm "msg"="Release reconciled" "gateway"={"Namespace":"kubesphere-controls-system","Name":"kubesphere-router-kubesphere-system"} "name"="kubesphere-router-kubesphere-system" "version"=1
I0927 15:28:38.348471       1 pvc.go:28] admitting pvc
I0927 15:28:39.399777       1 request.go:668] Waited for 1.043742475s due to client-side throttling, not priority and fairness, request: GET:https://10.96.0.1:443/apis/servicemesh.kubesphere.io/v1alpha2?timeout=32s
E0927 15:28:39.501562       1 pvc.go:65] get target Delete PVC from client failed, err:resource name may not be empty
I0927 15:28:44.583098       1 reconciler.go:682] controllers/Helm "msg"="Release reconciled" "nginx"={"Namespace":"kubesphere-controls-system","Name":"kubesphere-router-kubesphere-system-ingress"} "name"="kubesphere-router-kubesphere-system-ingress" "version"=1
I0927 15:29:28.416143       1 reconciler.go:682] controllers/Helm "msg"="Release reconciled" "gateway"={"Namespace":"kubesphere-controls-system","Name":"kubesphere-router-kubesphere-system"} "name"="kubesphere-router-kubesphere-system" "version"=1
I0927 15:29:39.651359       1 pvc.go:28] admitting pvc
I0927 15:29:40.702849       1 request.go:668] Waited for 1.044701224s due to client-side throttling, not priority and fairness, request: GET:https://10.96.0.1:443/apis/servicemesh.kubesphere.io/v1alpha2?timeout=32s
E0927 15:29:40.804484       1 pvc.go:65] get target Delete PVC from client failed, err:resource name may not be empty
I0927 15:29:44.932395       1 reconciler.go:682] controllers/Helm "msg"="Release reconciled" "nginx"={"Namespace":"kubesphere-controls-system","Name":"kubesphere-router-kubesphere-system-ingress"} "name"="kubesphere-router-kubesphere-system-ingress" "version"=1
```

**Comment by shou55** [comment]
+1  ,   I had the same problem

**Comment by guolong123** [comment]
这个问题有解决吗？

**Comment by stoneshi-yunify** [comment]
fixed by https://github.com/kubesphere/storageclass-accessor/commit/8397922fe5cf0441ac920024b2228fcb69c67ee9

**Comment by guolong123** [comment]
该问题在最新版本的镜像中仍存在，请确认下：
<img width="1284" alt="image" src="https://user-images.githubusercontent.com/22833963/229491484-e3e7f26e-2c6a-49a6-9781-7b32a9dfd7df.png">
参考：https://github.com/kubesphere/kubesphere/issues/5620#issuecomment-1494111569

**Comment by guolong123** [comment]
> fixed by [8397922](https://github.com/kubesphere/storageclass-accessor/commit/8397922fe5cf0441ac920024b2228fcb69c67ee9)
